jax/_src/api.py

```diff
@@ -228,15 +229,16 @@ def _infer_argnums_and_argnames(
 
 
 def jit(
-  fun: Callable,
-  *,
-  static_argnums: Union[int, Iterable[int], None] = None,
-  static_argnames: Union[str, Iterable[str], None] = None,
-  device: Optional[xc.Device] = None,
-  backend: Optional[str] = None,
-  donate_argnums: Union[int, Iterable[int]] = (),
-  inline: bool = False,
-) -> stages.Wrapped:
+    fun: Callable,
+    *,
+    static_argnums: Union[int, Iterable[int], None] = None,
+    static_argnames: Union[str, Iterable[str], None] = None,
+    device: Optional[xc.Device] = None,
+    backend: Optional[str] = None,
+    donate_argnums: Union[int, Iterable[int]] = (),
+    inline: bool = False,
+    abstracted_axes: Optional[Any] = None,
```

Comment:
I can't quite tell how is the abstract_axes supposed to work. I am guessing that this can be a sequence of specifications, one for each argument. Each argument specification can be a pytree matching the argument type. And at the leaves, we have a sequence of strings?

E.g., how to we specify that the 2nd and 3rd dimensions have abstract size and are equal for a function that takes an array with 3 dimensions?
```
abstract_axes=[(?, 'n', 'n')]
```

Reply:
> I can't quite tell how is the abstract_axes supposed to work.

Sounds like the lack of documentation is working as intended! 😆

Reply:
(We talked about this in a 1:1 meeting earlier today. The API is just a placeholder now.)